helpers/dateTime.js

```diff
@@ -55,6 +55,24 @@ export function formatDateInISOTime(val) {
 	return `${hours}:${minutes}:${seconds}`;
 }
 
+export function getClosestValidDate(minValue, maxValue, dateTime) {
+	const today = getToday();
+	const todayDate = getDateFromDateObj(today);
+	const minDate = dateTime ? getDateFromISODateTime(minValue) : getDateFromISODate(minValue);
+	const maxDate = dateTime ? getDateFromISODateTime(maxValue) : getDateFromISODate(maxValue);
+	if (isDateInRange(todayDate, minDate, maxDate)) {
+		return dateTime ? formatDateTimeInISO(convertLocalToUTCDateTime(today)) : formatDateInISO(today);
+	} else {
+		if (minValue && maxValue) {
+			const diffToMin = Math.abs(todayDate.getTime() - minDate.getTime());
+			const diffToMax = Math.abs(todayDate.getTime() - maxDate.getTime());
+			if (diffToMin < diffToMax) return minValue;
+			else return maxValue;
+		} else if (minValue) return minValue;
+		else if (maxValue) return maxValue;
+	}
```

Comment:
I'm guessing this never returns `undefined` because implicitly if there's no min/max specified, `isDateInRange` will always return `true`. Not sure if there's a way to make that more clear looking at the code (I'm surprised the linter isn't complaining).

Reply:
Yup exactly - Good idea, I'll try to make it more clear within this function.